src/cpu/flexcpu/simple_dataflow_thread.hh

```diff
@@ -466,6 +475,51 @@ class SDCPUThread : public ThreadContext
      */
     void predictCtrlInst(std::shared_ptr<InflightInst> inst_ptr);
 
+    /**
+     * Utility function for making a request for the CPU to do the memory
+     * access for this instruction. Should be called once no dependencies
+     * remain for the memory stage of the instruction. Will possibly be a
+     * speculative load, in which case squash logic will be necessary.
+     *
+     * std::weak_ptr<InflightInst> variant does a check before performing task.
+     *
+     * @param inst A reference to the instruction for which to request memory.
+     * @param req The request object which non-split requests will use.
+     * @param write Whether the request is a write. If not write, then read.
+     * @param data The data to write, if a write is requested.
+     * @param sreq A split request structure containing the divided and
+     *  translated requests.
+     */
+    void sendToMemory(std::weak_ptr<InflightInst> inst,
+                      const RequestPtr& req, bool write,
+                      std::shared_ptr<uint8_t> data,
```

Comment:
Could this have been a `vector<uint8_t>`. Seems like that would be safer since it's a pointer to data, not a single `uint8_t`.

Reply:
:T well you'd be copying that data twice in the case of a split access, and multiple times since we're passing it by value through lambdas in order to allow it to survive to calls made in different ticks...